From 7a2d9767585ed2c407d4985bd2d81552034fb90a Mon Sep 17 00:00:00 2001
From: CHEN Xiangyu <xiangyu.chen@aol.com>
Date: Thu, 9 Feb 2023 18:41:52 +0800
Subject: [PATCH] privsep-linux: fix SECCOMP_AUDIT_ARCH missing ppc64le (#181)

when dhcpcd running on ppc64le platform, it would be killed by SIGSYS.

Upstream-Status: Backport [7a2d9767585ed2c407d4985bd2d81552034fb90a]

Signed-off-by: Xiangyu Chen <xiangyu.chen@windriver.com>
---
 src/privsep-linux.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/privsep-linux.c b/src/privsep-linux.c
index 7372d26b..6a301950 100644
--- a/src/privsep-linux.c
+++ b/src/privsep-linux.c
@@ -232,7 +232,11 @@ ps_root_sendnetlink(struct dhcpcd_ctx *ctx, int protocol, struct msghdr *msg)
 #elif defined(__or1k__)
 #  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_OPENRISC
 #elif defined(__powerpc64__)
-#  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_PPC64
+#  if (BYTE_ORDER == LITTLE_ENDIAN)
+#    define SECCOMP_AUDIT_ARCH AUDIT_ARCH_PPC64LE
+#  else
+#    define SECCOMP_AUDIT_ARCH AUDIT_ARCH_PPC64
+#  endif
 #elif defined(__powerpc__)
 #  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_PPC
 #elif defined(__riscv)
-- 
2.34.1

